### ISSUE TITLE
user defined ROOTSYS ignored

### ISSUE THREAD

**Issue by theoreticMK** [rootpy/root_numpy#175]
Hi,
I have a system with 2 versions of ROOT installed. Older one in /usr/local provided by package manager, and a new one installed from source in /opt/root5 which I use by default (environment setup in .bashrc).

_librootnumpy.so linked against the old version regardless of my settings
I tried the suggested: 
```bash
sudo ROOTSYS=$ROOTSYS python setup.py install
```
which did not work

I tracked down the reason to the setup.py, line 47-48:
```python
try:
    root_cflags, root_ldflags = root_flags()
```
This line finds the old version of root-config in PATH.
As a quick fix I put '/opt/root5/bin/root-config' as argument to the function. The flags are then correct.

In summary: in setup.py ROOTSYS environment variable does not take precedence over default ROOT installation.

p.s. my old version of root 5.34/00 resulted in the following error when trying to use function tree2rec:
*** glibc detected *** python: double free or corruption
new ROOT version 5.34/25 works fine

**Comment by kratsg** [comment]
Hi, nobody commented on this. @ndawe, I'm thinking to allow for user precedence, we could provide a flag...

Relevant code here: https://github.com/rootpy/root_numpy/blob/master/setup.py#L47-48

```python
try:
  root_config = os.getenv('RNP_ROOT_CONFIG', 'root-config'):
  root_cflags, root_ldflags = root_flags(root_config)
except OSError:
  # ....
```

which lets users specify a flag so one could do perhaps

```RNP_ROOT_CONFIG=/opt/root5/bin/root-config pip install root_numpy```



> p.s. my old version of root 5.34/00 resulted in the following error when trying to use function tree2rec:
> *** glibc detected *** python: double free or corruption
> new ROOT version 5.34/25 works fine

As for this portion, I don't believe we supported 5.34/00 . I remember using PyROOT for that version and it was hella buggy. Please use 5.34/18 or 5.34/25. Otherwise, start looking into ROOT6.

**Comment by ndawe** [comment]
Ah, sorry. Catching up on this issue... 

> It does bother me that your $ROOTSYS doesn't quite point (or align) with the location of root-config which should be appropriately aliased in your path.

Agreed. $ROOTSYS and $PATH should be consistent.

But should be just change setup.py to give priority to ROOTSYS if it is set? It currently uses ROOTSYS as a fallback.

**Comment by theoreticMK** [comment]
Hi, thanks. Just to be clear, my user setup $ROOTSYS and $PATH are o.k.
ROOTSYS=/opt/root5
PATH=/opt/root5/bin:/usr/local/bin:/usr/bin
so when running root-config from shell I use the correct version of root. The problem arised only when running the setup.py as described above. The reason is that when running as sudo the PATH does not include the /opt/root5. 

Now when I know the reason it is easy to fix, but it took some time to find it. In my opinion I would give ROOTSYS priority.

**Comment by kratsg** [comment]
@theoreticMK - why are you running as sudo? Secondly, I tried this myself and I have no issues with the path if I attempt to install via sudo. Are you sure your setup is totally ok?

**Comment by theoreticMK** [comment]
@kratsg I just followed the installation instructions from http://rootpy.github.io/root_numpy/install.html and I wanted to install root_numpy system-wide. I am not sure that my setup is completely OK. What I do is that I have the following line in my .bashrc. 
```bash
source /opt/root5/bin/thisroot.sh
``` 
If there is something I miss, please let me know.

**Comment by kratsg** [comment]
@theoreticMK - it is not a good idea to install anything python-related system-wide. Use a virtual environment to wrap blocks of packages: https://virtualenvwrapper.readthedocs.org/en/latest/

As this allows you to maintain multiple versions and it works a lot better with ROOT. This **is** the official recommendation of the ROOT devs with PyROOT.

**Comment by theoreticMK** [comment]
@kratsg Thanks for the recommendation. It would be nice to have this in the documentation. Maybe with some short example how to install root+rootpy+root_numpy.

**Comment by kratsg** [comment]
@theoreticMK - it's not for us to add to be honest. When you install ROOT, it looks at your python version and compiles libraries and links PyROOT appropriately. `numpy` also needs to match up to the same Python version here. And `root_numpy` requires the PyROOT extensions (basically ROOT) and `numpy` which requires that it was installed with the same Python and compilers that existed when you installed ROOT.

The short example would be something like
```
brew install root
pip install virtualenvwrapper
echo "export WORKON_HOME=$HOME/.virtualenvs
export PROJECT_HOME=$HOME/Devel
source /usr/local/bin/virtualenvwrapper.sh
" >> ~/.bashrc
. ~/.bashrc
mkvirtualenv root
pip install numpy
pip install root_numpy
```

then whenever you open up terminal and want to use `root_numpy`, just need to type

```
workon root
```

All of this is from the docs I linked you basically (https://virtualenvwrapper.readthedocs.org/en/latest/install.html#python-versions) as well as how I normally install ROOT.

**Comment by theoreticMK** [comment]
Thanks again. I am new to using python for more than just simple scripts, so I find every advice very helpful.

**Comment by ndawe** [comment]
ROOTSYS now has priority if it is set. This is done in #187. Closing this issue now.
